src/bin/pg_autoctl/file_utils.c

```diff
@@ -764,3 +764,24 @@ fformat(FILE *stream, const char *fmt, ...)
 	va_end(args);
 	return len;
 }
+
+
+/*
+ * sformat is a secured down version of pg_snprintf
+ */
+int
+sformat(char *str, size_t count, const char *fmt, ...)
+{
+	int			len;
+	va_list		args;
+
+	if (str == NULL || fmt == NULL)
+	{
```

Comment:
probably good to log here, these should both never happen

Reply:
Log added, thanks!